src/index.ts

```diff
@@ -311,6 +311,32 @@ export default class Java8SpringBoot2Target extends Target {
             return options.inverse(this);
         });
 
+        engine.registerHelper('params', function(this: any) {
+            let argument = undefined;
+            let optional = false;
+
+            const body = 'transport' in this && this['transport'].toUpperCase() === "BODY";
+            const header = 'transport' in this && this['transport'].toUpperCase() === "HEADER";
+
+            if ('argumentName' in this && !body && !header) {
+                argument = '"' + this['argumentName'] + '"';
+            }
+            if ('argument' in this && header) {
+                argument = '"' + this['argument'] + '"';
+            }
+
+            if ('optional' in this && this['optional']) {
+                optional = true;
+            }
+
+            if (argument) {
+                let formatted = optional ? 'name = ' + argument + ', required = false' : argument;
+                return '(' + formatted + ')';
+            }
+
+            return "";
```

Comment:
What if no "argument" but it's optional ? Maybe that doesnt happen?

Reply:
It does. This is now fixed.